src/tests/java/htsjdk/samtools/util/SamLocusIteratorTest.java

```diff
@@ -64,6 +64,34 @@ public void testBasicIterator() {
         for (final SamLocusIterator.LocusInfo li : sli) {
             Assert.assertEquals(pos++, li.getPosition());
             Assert.assertEquals(2, li.getRecordAndPositions().size());
+            // make sure that we are not accumulating indels
+            Assert.assertEquals(li.getDeletedInRecord().size(), 0);
+            Assert.assertEquals(li.getInsertionInRecord().size(), 0);
+        }
+
+    }
+
+    @Test
+    public void testBasicIteratorWithIndels() {
+
+        final String sqHeader = "@HD\tSO:coordinate\tVN:1.0\n@SQ\tSN:chrM\tAS:HG18\tLN:100000\n";
```

Comment:
I've just followed the pattern found to keep the class without many changes, but I could have a look to `SAMRecordSetBuilder` and use it instead. In addition, I also found that some of the `Asser.assertEquals()` from the original class are in the wrong order (expected/actual instead of actual/expected). Should I change that too?